lib/probe.ml

```diff
@@ -0,0 +1,352 @@
+(*
+   Inputs:
+   - Loader.db (for creating probes, etc.)
+   - Names of unique RRs
+   - Received packets
+   - Event: application initiates first probe
+   - Event: delay complete
+   - Event: send complete
+   - (To Do: send fail or network down? trie changed?)
+
+   Outputs (actions):
+   - Packets to be sent
+   - Delay time
+   - Idle/finished indications
+*)
+
+let multicast_ip = Ipaddr.V4.of_string_exn "224.0.0.251"
```

Comment:
this might be obvious, but any chance of making this `multicast_dns_ip`?  I had to look it up to make sure this was the case...